charm/tests/unit/test_charm.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2023 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Unit tests for Testflinger Juju charm
+"""
+
+from pathlib import Path
+import pytest
+import yaml
+
+import ops
+import ops.testing
+from charm import TestflingerCharm
+
+CHARMDIR = Path(__file__).parent.parent.parent.resolve()
+METADATA_FILE = CHARMDIR / "metadata.yaml"
+# METADATA = str(yaml.safe_load(Path("metadata.yaml").read_text()))
+METADATA = str(yaml.safe_load(METADATA_FILE.read_text()))
+
+
+@pytest.fixture(name="harness")
+def fixture_harness():
+    """Pytest fixture for getting the harness"""
+    harness = ops.testing.Harness(TestflingerCharm, meta=METADATA)
+    return harness
+
+
+def test_waiting_for_relation(harness):
+    """Test that the status is set to waiting when relation is not present"""
+    harness.begin()
+    with pytest.raises(SystemExit):
+        assert harness.container_pebble_ready("testflinger")
+    assert isinstance(harness.model.unit.status, ops.WaitingStatus)
+
+
+def test_good_relation_active_status(harness):
+    """status should be active when the relation is present"""
+    harness.container_pebble_ready("testflinger")
+    harness.set_can_connect("testflinger", True)
```

Comment:
I'm wondering if it is worth making "testflinger" a constant (e.g. `PROJECT_NAME`) bc if you change it (for instance, make it uppercase or change spelling for some reason, you'll have to do it in all the places in the project

Reply:
I think it actually maps to the "PEBBLE_SERVICE_NAME" from a juju perspective if you look at the charm. I suppose if we were going to do that, we would define it in the charm and import it from there. However I'm not really sure that adds anything to the clarity (for me at least). Also, if we ever change the name from "testflinger" to something else, there will be a LOT more changes than this that are needed.

I'm not opposed to it though. If you think it makes it more readable for you, I'm happy to change it. Just that for me, I don't think it helps since I would need to go figure out what the value of that is whenever I see it.

Reply:
if you're okay to keep it, we can keep it :)